### PR TITLE
Define a deterministic ordering for ConfigMap and Secret ConfigSources

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java [hunk 1]
```diff
@@ -79,6 +81,17 @@ abstract ConfigSource createPropertiesConfigSource(String kubernetesConfigSource
         return result;
     }
 
+    private int getOrdinal(int ordinalOffset) {
+        final OrdinalData ordinalData = ordinalData();
+        /*
+         * We don't want a large list of sources to cause an "overflow" into an Ordinal of another ConfigSource
+         * so we just let the last ones all use the max ordinal
+         * this is not fool proof, but it's very unlikely that an application will need to define
+         * a list with 10+ sources...
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java:90]: Haha, I was thinking about this too :-) I think you're right here.

Comment (geoand) [reply extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java:90]: Glad to hear it :)

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java [hunk 2]
```diff
@@ -23,14 +23,20 @@
     public boolean failOnMissingConfig;
 
     /**
-     * ConfigMaps to look for in the namespace that the Kubernetes Client has been configured for
+     * ConfigMaps to look for in the namespace that the Kubernetes Client has been configured for.
+     * ConfigMaps defined later in this last have a higher priority that ConfigMaps defined earlier
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java:27]: last -> list

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java [hunk 3]
```diff
@@ -23,14 +23,20 @@
...
+     * in this list.
+     * Furthermore any Secrets defined in {@code secrets}, will have higher priorities than all ConfigMaps.
      */
     @ConfigItem
     public Optional<List<String>> configMaps;
 
     /**
      * Secrets to look for in the namespace that the Kubernetes Client has been configured for.
      * If you use this, you probably want to enable {@code quarkus.kubernetes-config.secrets.enabled}.
+     * Secrets defined later in this last have a higher priority that ConfigMaps defined earlier
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java:37]: last -> list

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/OrdinalData.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package io.quarkus.kubernetes.client.runtime;
+
+final class OrdinalData {
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/OrdinalData.java:3]: I'd make this an `enum`, but that's just me.

**Comment by Ladicek** [comment]
LGTM, just spotted a few typos in the docs.

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/OrdinalData.java (hunk 4 above)
Comment (geoand) [reply extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/OrdinalData.java:3]: Good point, I'll fix it

**Comment by geoand** [comment]
PR updated per suggestions

**Review by Ladicek — APPROVED** [review]

**Pull request merged**